### PR TITLE
add support for azure resources integration

### PR DESCRIPTION
**Pull request by taimoorgit**
## Issues

https://github.com/OpsLevel/team-platform/issues/414

## Changelog

- [x] Add azure resources graphql fragment
- [x] Add azure resources new client functions 
- [x] Update integrations.go tests
- [x] Update cache.go tests
- [x] Make a `changie` entry

## Tophatting

Running create integration query on staging3:

image

Running update integration query on staging3:

image

### CODE REVIEW

File: integration.go [hunk 1]
```diff
@@ -19,6 +21,15 @@ type AWSIntegrationFragment struct {
 	OwnershipTagKeys     []string `graphql:"ownershipTagKeys"`
 }
 
+type AzureResourcesIntegrationFragment struct {
+	TenantId                                             string        `graphql:"tenantId"`
+	SubscriptionId                                       string        `graphql:"subscriptionId"`
+	LastSyncedAt                                         *iso8601.Time `graphql:"lastSyncedAt"`
+	Aliases                                              []string      `graphql:"aliases"`
+	AllowManualSyncInfrastructureResources               bool          `graphql:"allowManualSyncInfrastructureResources"`
+	MinutesUntilManualSyncInfrastructureResourcesAllowed *int          `graphql:"minutesUntilManualSyncInfrastructureResourcesAllowed"`
```
Comment (rocktavious) [inline integration.go:30]: are these actually going to be public graphql fields?

Comment (taimoorgit) [reply integration.go:30]: Cannot really tell from the graphql schema file, so I am reaching out to confirm.

Comment (Farjaad) [reply integration.go:30]: everything but `allowManualSyncInfrastructureResources` and  `minutesUntilManualSyncInfrastructureResourcesAllowed` will be in the public schema

Comment (rocktavious) [reply integration.go:30]: ok cool - make sure we only add whats going to be in the public schema

**Review by rocktavious — APPROVED** [review]
Great Work @taimoorgit

**Review by davidbloss — APPROVED** [review]
LGTM! 🎉

**Review by rocktavious — APPROVED** [review]
LGTM

**Pull request merged**